firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/LocalOverrideSettings.kt

```diff
@@ -18,59 +18,57 @@ package com.google.firebase.sessions.settings
 
 import android.content.Context
 import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
 import kotlin.time.Duration
 import kotlin.time.DurationUnit
 import kotlin.time.toDuration
 
-internal class LocalOverrideSettings(val context: Context) : SettingsProvider {
-
-  private val sessions_metadata_flag_sessionsEnabled = "firebase_sessions_enabled"
-  private val sessions_metadata_flag_sessionRestartTimeout =
-    "firebase_sessions_sessions_restart_timeout"
-  private val sessions_metadata_flag_samplingRate = "firebase_sessions_sampling_rate"
+internal class LocalOverrideSettings(context: Context) : SettingsProvider {
   private val metadata =
-    context.packageManager
-      .getApplicationInfo(context.packageName, PackageManager.GET_META_DATA)
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        context.packageManager.getApplicationInfo(
+          context.packageName,
+          PackageManager.ApplicationInfoFlags.of(PackageManager.GET_META_DATA.toLong()),
+        )
+      } else {
+        @Suppress("DEPRECATION") // For older API levels.
+        context.packageManager.getApplicationInfo(
+          context.packageName,
+          PackageManager.GET_META_DATA,
+        )
+      }
```

Comment:
It's a bit hard to clean this up because if I do any other than inline everything like this, I run into https://youtrack.jetbrains.com/issue/KT-53650 

I tried to make it clearer, I moved the .metaData inside the if and else bodies, and added a comment for the default value.